src/validationinterface.h

```diff
@@ -34,6 +35,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {}
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {}
     virtual void BlockChecked(const CBlock&, const CValidationState&) {}
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
Lets say if we support two wallets, we could define which wallet takes the signaling-lead by setting the signal registration order (last signal connect = last signal call).

Throwing an error because both wallets could provide a script for mining seems to be relatively strict. The current solution would allow the 2nd (or upcoming) wallet(s) to decide if he likes to take the lead and overwrite or allow the first wallets script.

Adding a combiner would mean that we somehow had to check, which wallet provided which script. Also in that case the miner is in charge to decide which script should be taken.